docs/src/references/tedge-p11-server.md

```diff
@@ -0,0 +1,163 @@
+draft: true
+---
+
+## Description
+
+**tedge-p11-server** is an optional component used by %%te%% for accessing PKCS #11 cryptographic
+tokens. It should be used when it's not possible to load the PKCS #11 dynamic module in `tedge`
+directly at runtime, e.g. when running a statically linked %%te%% on musl or when `tedge` doesn't
+have the permission to view/access cryptographic tokens.
+
+When running `tedge connect` or `tedge reconnect` command, as part of a TLS handshake with the
+remote MQTT broker, a confirmation of an ownership of the certificate by the client is required.
+This is achieved by signing a TLS 1.3 CertificateVerify message by the PKCS #11 cryptographic token.
+This happens only once when establishing an MQTT connection over TLS and will only need to be
+repeated when a new connection is opened.
+
+If you can load dynamic objects at runtime and can access the token directly, you can instead use a
+`module` cryptoki mode, where the module is used directly in tedge.
+<!--TODO: link how to use module mode-->
+
+<!--TODO: link to an appropriate section of how to select a token using the URI in the HSM support document-->
+For information about how to select appropriate token/object, see [HSM reference page](./hsm-support.md).
+
+## Connecting to the server from tedge
+
+`tedge-p11-server` exposes a UNIX socket `tedge` will connect to to verify the client certificate
+when connecting to the cloud. To make `tedge` use `tedge-p11-server` for client authentication, it's
+necessary to set the mode to `socket`:
+
+```shell
+tedge config set device.cryptoki.mode socket
+tedge config set device.cryptoki.socket_path /path/to/socket.sock # default: /run/tedge-p11-server/tedge-p11-server.sock
+```
+
+## Running the server
+
+### Method 1: Invoking the server from the CLI
+
+To make sure everything works, you can start the server manually from the CLI:
+
+```shell
+tedge-p11-server --module-path /path/to/pkcs11-module.so
+```
+```sh title="Output"
+2025-05-06T11:13:40.171073Z  INFO tedge_p11_server: Using cryptoki configuration cryptoki_config=CryptokiConfigDirect { module_path: "/path/to/pkcs11-module.so", pin: "[REDACTED]", uri: None }
+2025-05-06T11:12:21.748700Z  INFO tedge_p11_server: Server listening listener=Some("./tedge-p11-server.sock")
+```
+
+`tedge-p11-server` will create the socket and delete it when it exits.
+
+### Method 2: Using systemd socket activation
+
+When using systemd, we can make it start the server when a client tries connecting to the socket
+instead of starting it manually.
+
+First, create the required `.service` and `.socket` files using `systemctl edit`:
+
+```sh title="systemctl edit --full tedge-p11-server.socket"
+[Unit]
+Description=tedge-p11-server socket
+
+[Socket]
+Priority=6
+Backlog=5
+ListenStream=%t/tedge-p11-server/tedge-p11-server.sock
+SocketUser=tedge
+SocketGroup=tedge
+SocketMode=0660
+
+[Install]
+WantedBy=sockets.target
+```
```

Comment:
We probably don't need to go into details about the actual unit definitions as they are part of the tedge-p11-server deb and rpm package. We can mention the services and how to use them, but we don't need to do `systemctl edit tedge-p11-server*` as everything can be configured using tedge (or manually editing the /etc/tedge/tedge.toml)

Reply:
Yeah, I copy-pasted the definitions unaware that they're now present in the `tedge-p11-server` package and the user only needs to install the package and enable the socket. The unit definitions will be removed.

Reply:
changed in 22324daa

Reply:
fixed in f8db9a50aff1021ee8fe70bbff7c0931184e172e